golem-worker-executor-base/src/cassandra.rs

```diff
@@ -0,0 +1,678 @@
+        }
+    }
+
+    pub async fn configured(config: &CassandraConfig) -> Result<Self, String> {
+        let mut session_builder = SessionBuilder::new()
+            .known_nodes_addr(config.nodes.iter())
+            .pool_size(PoolSize::PerHost(
+                NonZeroUsize::new(config.pool_size_per_host).unwrap(),
+            ))
+            .use_keyspace(&config.keyspace, false);
+
+        if let (Some(username), Some(password)) =
+            (config.username.as_ref(), config.password.as_ref())
+        {
+            session_builder = session_builder.user(username, password);
+        }
+
+        let session = session_builder.build().await.map_err(|e| e.to_string())?;
+
+        Ok(CassandraSession {
+            session: Arc::new(session),
+            keyspace: config.keyspace.clone(),
+            set_tracing: config.tracing,
+        })
+    }
+
+    pub async fn create_docker_schema(&self) -> Result<(), String> {
+        self.session.query_unpaged(
+            Query::new(
+                format!("CREATE KEYSPACE IF NOT EXISTS {} WITH REPLICATION = {{ 'class' : 'SimpleStrategy', 'replication_factor' : 1 }};", self.keyspace),
+            ),
+            &[],
+        ).await
+            .map_err(|e| e.to_string())?;
+
+        self.session
+            .query_unpaged(
+                Query::new(format!(
+                    r#"
+            CREATE TABLE IF NOT EXISTS {}.kv_store (
+                namespace TEXT,
+                key TEXT,
+                value BLOB,
+                PRIMARY KEY (namespace, key)
+            );
+        "#,
+                    self.keyspace
+                )),
+                &[],
+            )
+            .await
+            .map_err(|e| e.to_string())?;
+
+        self.session
+            .query_unpaged(
+                Query::new(format!(
+                    r#"
+            CREATE TABLE IF NOT EXISTS {}.kv_sets (
+                namespace TEXT,
+                key TEXT,
+                value BLOB,
+                PRIMARY KEY ((namespace, key), value)
+            );
+        "#,
+                    self.keyspace
+                )),
+                &[],
+            )
+            .await
+            .map_err(|e| e.to_string())?;
+
+        self.session
+            .query_unpaged(
+                Query::new(format!(
+                    r#"
+                CREATE TABLE IF NOT EXISTS {}.kv_sorted_sets (
+                    namespace TEXT,
+                    key TEXT,
+                    score DOUBLE,
+                    value BLOB,
+                    PRIMARY KEY ((namespace, key), score, value)
+                );
+        "#,
+                    self.keyspace
+                )),
+                &[],
+            )
+            .await
+            .map_err(|e| e.to_string())
+            .map(|_| ())
+    }
+
+    pub fn with(&self, svc_name: &'static str, api_name: &'static str) -> CassandraLabelledApi {
+        CassandraLabelledApi {
```

Comment:
What is the intended relation between CassandraSession, CassandraLabelledApi, CassandraKeyValueStorage?

iIm a bit confused, as the CassandraSession and CassandraLabelledApi suggest some generic functionality, but AFAIU all the specific key values storage implementation is in CassandraLabelledApi, which wraps a CassandraSession, which contains the specific key-value schema, and CassandraKeyValueStorage is calling CassandraSession, and wrapping it before every call into a CassandraLabelledApi.

Maybe I'm missing something, but
 - I see no reason for a separate CassandraSession, the schema is tied to CassandraKeyValueStorage
 - I think the work done by CassandraLabelledApi could be a generic wrapper method in CassandraKeyValueStorage, instead of duplicating all the API
 - and I think the queries should live in CassandraKeyValueStorage

Reply:
So I do see that we have in RedisLabbelled in common, but that is not "IndexedKeyValue" level, rather a generic redis wrapper (ofc because redis is a key value too in general, the two layer matches somewhat). I think we can have CassandraLabelled, but that should be in common, and on the level of cassandra operations (e.g. execute query with the common labels and a custom "query-name" label)

Reply:
@vigoo wdty?

Reply:
The problem to move CassandraLabelled into common is that we are going to have new dependencies in CLI, which I remember we want to avoid, that's why I kept everything in executor-worker modules until the refactor comes to be able to move all common stuff into common without impacting CLI or others.

Reply:
well, we also have service base and / or we can use features.
which refactor is that?

Reply:
The redis code in common adds metrics and logging to the redis client and it just directly wraps the redis operations, independent of any actual use case.

I see that in this top-level cassandra module there is key-value store specific queries, which is definitely not something we want. Everything key-value store specific should be in the `storage/keyvalue/cassandra` module. 

If you want to have common metrics and logging for cassandra, like we have for redis, then you can wrap the cassandra library but that should just instrument (wrap) the library's functions and not add any logic on top if it.

Where this "common" cassandra wrapper is is another question, top-level worker-executor-base is definitely not a good place for it. Until we need it in any other service, we can keep it in worker-executor-base, but let's move it to the storage module at least.